testsuite/cluster-test/src/atomic_histogram.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::tx_emitter::TxStats;
+use std::sync::atomic::{AtomicU64, Ordering};
+
+const HISTOGRAM_CAPACITY: usize = 2048;
+const HISTOGRAM_STEP_MILLIS: u64 = 50;
+
+pub struct AtomicHistogram {
+    capacity: usize,
+    step_millis: u64,
+    buckets: Vec<AtomicU64>,
+}
+
+impl Default for AtomicHistogram {
+    fn default() -> AtomicHistogram {
+        let mut buf = Vec::with_capacity(HISTOGRAM_CAPACITY);
+        for _i in 0..HISTOGRAM_CAPACITY {
+            buf.push(AtomicU64::new(0));
+        }
+        Self {
+            capacity: HISTOGRAM_CAPACITY,
+            step_millis: HISTOGRAM_STEP_MILLIS,
+            buckets: buf,
+        }
+    }
+}
+
+impl AtomicHistogram {
```

Comment:
Can you add `fn new(capacity, step_millis)` and call it from `default()`?
Also, can you rename HISTOGRAM_CAPACITY=>DEFAULT_ HISTOGRAM_CAPACITY and same for HISTOGRAM_STEP_MILLIS